### ISSUE TITLE
EverShifting-Fountain Freezes server

### ISSUE THREAD

**Issue by blakdragan7** [HellFirePvP/AstralSorcery#415]
I am running a descent server, over 32 gigs of ram 2x 8 core intel xenons at 3.9 ghz. But when I activate the evershifting fountain the minecraft server freezes and if i ssh to the server and run top I see that java is using basically all of 2 cpu cors. I'm assuming because minecraft only uses a few cores. I have literally nothing else running on the server except ssh-daemon of course. Is this a known issue ? I did a lot of googling around and couldn't find anything. I fixed it by just removing one of the blocks in the fountain multiblock but the error always comes back once I re activate the fountain. This seems like some sort of recursion problem maybe ? any help would be awesome. Thanks !

**Comment by blakdragan7** [comment]
forgot to mention mod version is 1.12.2-1.3.0 and It's in the DireWrlf20 pacj on 1.12.2

**Comment by blakdragan7** [comment]
If someone wouldn't mind giving me some guidance (as in, where is that code at) I could fix it myself if needed.

**Comment by Doomgull** [comment]
Server logs, please.

**Comment by blakdragan7** [comment]
[2017-12-03-1.log](https://github.com/HellFirePvP/AstralSorcery/files/1527747/2017-12-03-1.log)
[2017-12-03-2.log](https://github.com/HellFirePvP/AstralSorcery/files/1527748/2017-12-03-2.log)
[2017-12-03-3.log](https://github.com/HellFirePvP/AstralSorcery/files/1527750/2017-12-03-3.log)

here you go. Sorry I am usually better at remembering to upload things like that. The end of log 1 is when we started the block, log 2 is when we restarted the server to see if it helped but it didnt and i removed a block from the fountain and restarted the server, log 3 is when the server was started right afterwords without the fountain. 

Let me know if any more information is needed.

**Comment by blakdragan7** [comment]
Oh also forgot to mention, I added a couple of mods to the pack, Embers, anamania, openmodulerturrets mekanism and roots.

**Comment by blakdragan7** [comment]
if you want me to try it again without those mods I can as well. Also I tried it in single player with all those mods and it seemed to work but was very laggy at first and when I re connected to the srever afterwords (without restarting minecraft in between) it was very laggy like there was some kind of memory leak or something until after i restarted minecraft and then it was fine.

**Comment by blakdragan7** [comment]
My desktop is a windows 8.1 machine btw. in case that matters

**Comment by blakdragan7** [comment]
Sorry, Also forgot to mention, after we restarted the server without the fountain (log 3) there were four block on the corner of where the fountain dug to bedrock that when a block was placed it would instantly apear in your inventory again but it a different not hotbar spot. I had to restart once again to fix the issue. here is the log file for when it was restarted again.

[2017-12-03-4.log](https://github.com/HellFirePvP/AstralSorcery/files/1527794/2017-12-03-4.log)

**Comment by blakdragan7** [comment]
I'm assuming that issue is related to this problem

**Comment by Doomgull** [comment]
The server is constantly dropping ticks even before the fountain is completed.

And some serious advancement issue spam.

Still looking, but may be related to performance because of the constant tickdrops even before the fountain is involved.

**Comment by blakdragan7** [comment]
lol I love the label. Once I get home from work I will try it without the extra mods I added to the pack. I will post on here what I find as well. I didn't actually read through the logs myself or I would have probably done that already after noticing the tick thing. I generally try not to fix bugs when I'm not at work lol

**Comment by blakdragan7** [comment]
Also when you say advancement issue spam do you mean the playername moved wrongly prints ?

**Comment by Doomgull** [comment]
No, when server restarts in the log2/3/4: "[18:13:47] [Server thread/ERROR]: Parsing error loading built-in advancement xxxxxxxx"

The player moved wrongly/too quickly thing, are you using something on the server or a plugin that controls player flight/movement/anticheat/itembreaking?

**Comment by blakdragan7** [comment]
I'm not using anything other then the mods for dw20 pack and the few extra mods i listed a second ago. I do travel around with the slime sling a lot though which is probably where thats from.

**Comment by Doomgull** [comment]
Its literally only server load, then. Fountain is a little more computationally complex to begin with, combined with a server that's already struggling before it even fires off (all the tick skips), I think thats the issue we're seeing here.

**Comment by blakdragan7** [comment]
it's very possible. Though I do have a very beefy server so I'm not sure why it would be struggling ? I will look into it. If I find anything I will post it here.

**Comment by Doomgull** [comment]
If the world hasn't been fuly pre-generated, that can make a huge difference, same with tweaking java arguments.

May want to run a profiler on the jvm, as well, and see if maybe something is causing excessive load in the world (mob farm, machines, etc)

**Comment by blakdragan7** [comment]
Well to be honest I didn't know pre-generating the world was a thing. I will give that a try and I found https://github.com/MinimallyCorrect/TickProfiler to try and find whats taking up most of the ticks. I have a very large amount of experience programming but not a whole lot with minecraft I'm afraid and not a whole lot more with java. I usually use c / c++. I'm sure google will have the answers on how to do all that stuff though.

**Comment by HellFirePvP** [comment]
Any news on this?

**Comment by blakdragan7** [comment]
yes sorry, I've been traveling for work and didn't have time to comment on here. It works fine now I just had to set some settings for java to make the server run better so it was a tick rate issue. Here is the startup file I use now that fixed it. If someone uses this though they should probably change the memory because I'm using 6g min and max. 
Also I changed it from .sh to .txt so github would let me upload/.

[ServerStart.txt](https://github.com/HellFirePvP/AstralSorcery/files/1552142/ServerStart.txt)
